test/e2e/volume_provisioning.go

```diff
@@ -114,45 +156,170 @@ var _ = framework.KubeDescribe("Dynamic provisioning", func() {
 	})
 
 	framework.KubeDescribe("DynamicProvisioner", func() {
-		It("should create and delete persistent volumes [Slow]", func() {
-			framework.SkipUnlessProviderIs("openstack", "gce", "aws", "gke")
+		// This test checks that dynamic provisioning can provision a volume
+		// that can be used to persist data among pods.
+
+		tests := []storageClassTest{
+			{
+				"should provision SSD PD on GCE/GKE",
+				[]string{"gce", "gke"},
+				"kubernetes.io/gce-pd",
+				map[string]string{
+					"type": "pd-ssd",
+					// Check that GCE can parse "zone" parameter, however
+					// we can't create PDs in different than default zone
+					// as we don't know if we're running with Multizone=true
+					"zone": framework.TestContext.CloudConfig.Zone,
+				},
+				"1.5Gi",
+				"2Gi",
+				func(volume *api.PersistentVolume) error {
+					return checkGCEPD(volume, "pd-ssd")
+				},
+			},
+			{
+				"should provision HDD PD on GCE/GKE",
+				[]string{"gce", "gke"},
+				"kubernetes.io/gce-pd",
+				map[string]string{
+					"type": "pd-standard",
+				},
+				"1.5Gi",
+				"2Gi",
+				func(volume *api.PersistentVolume) error {
+					return checkGCEPD(volume, "pd-standard")
+				},
+			},
+			// AWS
+			{
+				"should provision gp2 EBS on AWS",
+				[]string{"aws"},
+				"kubernetes.io/aws-ebs",
+				map[string]string{
+					"type": "gp2",
+					// Check that AWS can parse "zone" parameter, however
+					// we can't create PDs in different than default zone
+					// as we don't know zone names
+					"zone": framework.TestContext.CloudConfig.Zone,
+				},
+				"1.5Gi",
+				"2Gi",
+				func(volume *api.PersistentVolume) error {
+					return checkAWSEBS(volume, "gp2", false)
+				},
+			},
+			{
+				"should provision io1 EBS on AWS",
+				[]string{"aws"},
+				"kubernetes.io/aws-ebs",
+				map[string]string{
+					"type":      "io1",
+					"iopsPerGB": "50",
+				},
+				"3.5Gi",
+				"4Gi", // 4 GiB is minimum for io1
+				func(volume *api.PersistentVolume) error {
+					return checkAWSEBS(volume, "io1", false)
+				},
+			},
+			{
+				"should provision sc1 EBS on AWS",
+				[]string{"aws"},
+				"kubernetes.io/aws-ebs",
+				map[string]string{
+					"type": "sc1",
+				},
+				"500Gi", // minimum for sc1
+				"500Gi",
+				func(volume *api.PersistentVolume) error {
+					return checkAWSEBS(volume, "sc1", false)
```

Comment:
Do you think it may be worth replacing this with `nil` like openstack checks because it doesn't do anything?